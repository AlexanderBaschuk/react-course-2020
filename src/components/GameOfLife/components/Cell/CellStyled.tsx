import styled from '@emotion/styled'

interface CellStyleProps {
	size: number
	isAlive: boolean
}

export const CellStyled = styled.div<CellStyleProps>`
	display: 'inline-block';
	width: ${(props) => props.size};
	height: ${(props) => props.size};
	backgroundColor: ${(props) => props.isAlive} ? '#000' : '#EEE'
`
