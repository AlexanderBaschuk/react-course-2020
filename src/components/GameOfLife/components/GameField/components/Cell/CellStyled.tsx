import styled from '@emotion/styled'

interface CellStyleProps {
	cellSize: number
	isAlive: boolean
	animate: boolean
	duration: number
}

export const CellStyled = styled.div<CellStyleProps>`
	display: inline-block;
	width: ${(props) => props.cellSize}px;
	height: ${(props) => props.cellSize}px;
	background-color: ${(props) => (props.isAlive ? '#000' : '#EEE')};
	transition: ${(props) => (props.animate ? `background-color ${props.duration}ms` : 'none')};
`
