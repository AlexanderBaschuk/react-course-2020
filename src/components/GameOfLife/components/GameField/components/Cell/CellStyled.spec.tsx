import renderer from 'react-test-renderer'
import { CellStyled } from './CellStyled'
import React from 'react'

test.each`
	cellSize | isAlive
	${10}    | ${true}
	${20}    | ${true}
	${20}    | ${false}
`('Cell snapshot test', ({ cellSize, isAlive }) => {
	const cell = renderer
		.create(<CellStyled cellSize={cellSize} isAlive={isAlive} />)
		.toJSON()
	expect(cell).toMatchSnapshot()
})
