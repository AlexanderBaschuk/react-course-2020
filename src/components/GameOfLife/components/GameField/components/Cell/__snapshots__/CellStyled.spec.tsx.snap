// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell snapshot test 1`] = `
.emotion-0 {
  display: inline-block;
  width: 10px;
  height: 10px;
  background-color: #000;
}

<div
  className="emotion-0"
/>
`;

exports[`Cell snapshot test 2`] = `
.emotion-0 {
  display: inline-block;
  width: 20px;
  height: 20px;
  background-color: #000;
}

<div
  className="emotion-0"
/>
`;

exports[`Cell snapshot test 3`] = `
.emotion-0 {
  display: inline-block;
  width: 20px;
  height: 20px;
  background-color: #EEE;
}

<div
  className="emotion-0"
/>
`;
